### PR TITLE
bootstrap: don't resolve symlinks for initial_cargo

### PR DESCRIPTION
**Pull request by GrigorenkoPV**
I have put the following in my `config.toml`:

```toml
# Includes one of the default files in src/bootstrap/defaults
profile = "compiler"
change-id = 121203

[build]
cargo = "/usr/bin/cargo"
rustc = "/usr/bin/rustc"
rustfmt = "/usr/bin/rustfmt"
```

I have rustup installed from Arch's repos, which has all of the above paths be symlinks to `/usr/bin/rustup`. This works just fine with the `argv[0]` trick that rustup uses.

However, `bootstrap` resolves symlinks to check whether `cargo` exists and then uses the resolved path, so it ends up calling `rustup` directly expecting it to behave like `cargo`. Which it doesn't.

This PR removes the canonicalization step, in turn fixing the issue, but sacrificing a pretty error message. However, this exact thing is checked by `x.py` in advance, so I hope it is not a big deal?

### CODE REVIEW

**Review by onur-ozkan — COMMENTED** [review]
Stopping bootstrapping on the invalid configuration is actually quite useful (so people doesn't waste time on building when configuration is invalid). I would rather change this validation to be running a simple cargo command and checking if it's working or not. With this approach, your usecase should be working just fine and we can keep validating the initial cargo.

**Comment by onur-ozkan** [comment]
r? onur-ozkan

File: src/bootstrap/src/core/config/config.rs [hunk 1]
```diff
@@ -1418,19 +1419,23 @@ impl Config {
...
-                t!(PathBuf::from(cargo).canonicalize(), "`initial_cargo` not found on disk")
-            })
-            .unwrap_or_else(|| config.out.join(config.build.triple).join("stage0/bin/cargo"));
+        config.initial_cargo = if let Some(cargo) = cargo {
+            if !flags.skip_stage0_validation {
+                config.check_stage0_version(&cargo, "cargo");
+            }
+            PathBuf::from(cargo)
+        } else {
+            config.download_beta_toolchain();
```
Comment (onur-ozkan) [inline src/bootstrap/src/core/config/config.rs:1436]: Oh, this fixes a bug. Thanks :)

**Review by onur-ozkan — APPROVED** [review]

**Comment by onur-ozkan** [comment]
I will take this to the merge queue once it's been rebased and the commits have been squashed/unified. Having the first commit no longer makes sense.

**Comment by GrigorenkoPV** [comment]
Thank you a lot for the review and the feedback!

Really appreciate it ❤️

**Pull request merged**